src/memgraph.cpp

```diff
@@ -1200,20 +1204,35 @@ int main(int argc, char **argv) {
                             []() -> nlohmann::json { return query::plan::CallProcedure::GetAndResetCounters(); });
   }
 
+  communication::websocket::Server websocket_server{
+      boost::asio::ip::tcp::endpoint{boost::asio::ip::make_address("0.0.0.0"), 7444}};
+
+  {
+    auto sinks = spdlog::default_logger()->sinks();
+    sinks.push_back(websocket_server.GetLoggingSink());
+    spdlog::set_default_logger(std::make_shared<spdlog::logger>("memgraph_log", sinks.begin(), sinks.end()));
+  }
+
   // Handler for regular termination signals
-  auto shutdown = [&server, &interpreter_context] {
+  auto shutdown = [&websocket_server, &server, &interpreter_context] {
     // Server needs to be shutdown first and then the database. This prevents
     // a race condition when a transaction is accepted during server shutdown.
     server.Shutdown();
     // After the server is notified to stop accepting and processing
     // connections we tell the execution engine to stop processing all pending
     // queries.
     query::Shutdown(&interpreter_context);
+    websocket_server.Shutdown();
   };
-  InitSignalHandlers(shutdown);
 
   MG_ASSERT(server.Start(), "Couldn't start the Bolt server!");
+  websocket_server.Start();
```

Comment:
Do we have websocket server as optional? :thinking:

Reply:
If we have the server compiled, we should enable it by default (otherwise, it's just consuming the memory and people have to make additional effort to start it). Being optional vs. what's the default value are two different questions. From my perspective, the more exciting thing is to make the server "compile-time option" for the environments where you have less memory and don't need WebSockets (I'm not sure what the binary size / actual memory consumption is). That seems to me as an exciting option 😄 But, even though that's interesting, I would ignore that for now 😂

Reply:
I agree with the compile out option, something we can consider if it proves too heavy for some env even though I doubt that. 
My question was for the assert, what happens if it fails to bind to a port? 
Do we terminate as is the case for bolt port or do we just print out an error?

Reply:
Aha, that's another point. Since WebSockets are not required for Memgraph to be operational, I think it makes more sense to print out an error and move on 🤔

Reply:
But yea, if the WebSockets-related code can deal with that? If no, than we have to assert here 😄